core/bones/key.py

```diff
@@ -21,7 +23,9 @@ def __init__(
         self.allowed_kinds = allowed_kinds
         self.check = check
 
-    def singleValueFromClient(self, value, skel, name, origData):
+    def singleValueFromClient(self, value: Any, skel: 'SkeletonInstance',
+                              bone_name: str, client_data: dict
+                              ) -> tuple[Any, list[ReadFromClientError] | None]:
```

Comment:
Is it necessary to provide full type annotations in sub-classes as well?
I would like to keep maintenance overhead low, if possible.